packages/strapi-plugin-content-type-builder/admin/src/containers/DataManagerProvider/reducer.js

```diff
@@ -458,6 +391,17 @@ const reducer = (state, action) => {
         }
       }
 
+      const attributes = state.getIn(['modifiedData', mainDataKey, 'schema', 'attributes']).toJS();
+      const uidField = Object.entries(attributes).find(
+        ([, value]) => attributeToRemoveName === value.targetField
+      );
+
+      if (uidField) {
+        return state
+          .removeIn(pathToAttributeToRemove)
+          .removeIn([...pathToAttributes, uidField[0], 'targetField']);
```

Comment:
Your algorithm works only if there is only one `uid` field attached to the removed `text` field.

In order to removed all the `targetField` fields attached to the removed one you need to:

1. Retrieve all the uid fields attached to the removed one:
```
const uidFieldsAttachedToRemovedField = Object.keys(attributes).filter(attributeName => {
  return attributeToRemoveName === attributes[attributeName].targetField;
});
```
2. Update the uid fields by removing the targetField key

```
return state.removeIn(pathToAttributeToRemove).updateIn([...pathToAttributes], attributes => {
  return attributes.keySeq().reduce((acc, current) => {

    if (uidFieldsAttachedToRemovedField.includes(current)) {
      return acc.removeIn([current, 'targetField']);
    }

    return acc;
  }, attributes);
});
```

> You don't need the if statement with this code